educator/src/Dscp/Educator/Web/Queries.hs

```diff
@@ -63,12 +64,13 @@ commonGetAssignments apiCase student filters = do
             EducatorCase ->
                 return AssignmentEducatorInfo{ aiHash = assignH, .. }
             StudentCase -> do
-                aiLastSubmission <- studentGetLastAssignmentSubmission student assignH
+                aiLastSubmission <- studentGetLastAssignmentSubmission (Unsafe.fromJust $ afStudent filters) assignH
```

Comment:
Its better to have two distinct functions.

Reply:
I would like to leave that for a cleanup PR after we're done fixing API to client requests.
@flyingleafe are you ok with that?